connector-common/src/main/java/com/redhat/cloud/notifications/connector/KafkaReinjectionProcessor.java

```diff
@@ -0,0 +1,51 @@
+package com.redhat.cloud.notifications.connector;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import org.apache.camel.Exchange;
+import org.apache.camel.Message;
+import org.apache.camel.Processor;
+
+import java.util.concurrent.TimeUnit;
+
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.KAFKA_REINJECTION_COUNT;
+import static com.redhat.cloud.notifications.connector.ExchangeProperty.KAFKA_REINJECTION_DELAY;
+import static com.redhat.cloud.notifications.connector.IncomingCloudEventFilter.X_RH_NOTIFICATIONS_CONNECTOR_HEADER;
+
+@ApplicationScoped
+public class KafkaReinjectionProcessor implements Processor {
+    @Inject
+    ConnectorConfig connectorConfig;
+
+    /**
+     * Determines the reinjection delay from the reinjection count which should
+     * have been extracted from the incoming message's reinjection header. If
+     * for some reason we don't have that count, we default that count to zero.
+     *
+     * Also, it sets the original Cloud Event as the exchange's message body.
+     * @param exchange the incoming exchange from the previous routes or error
+     *                 handlers.
+     */
+    @Override
+    public void process(final Exchange exchange) {
+        final int reinjectionCount = exchange.getProperty(KAFKA_REINJECTION_COUNT, 0, Integer.class);
+
+        final long reinjectionDelay = switch (reinjectionCount) {
+            case 0 -> TimeUnit.SECONDS.toMillis(10);
+            case 1 -> TimeUnit.SECONDS.toMillis(30);
+            case 2 -> TimeUnit.MINUTES.toMillis(1);
+            default -> this.connectorConfig.getIncomingKafkaMaxPollIntervalMs() / 2;
+        };
```

Comment:
Since the delayed reinjection happens with the data since in memory, we can't use a long delay value. Based on that, I wonder if we should make this simpler and only have one delay value set up from the configuration (and overridden from an env var). One minute would be a good default value.

Reply:
To be addressed at a later time.